xgcm/grid.py

```diff
@@ -1778,7 +1778,8 @@ def _1d_grid_ufunc_dispatch(
         else:
             dask = "forbidden"
 
-        array = da
+        # Need to copy to avoid modifying in-place. Ideally we would test for this behaviour specifically
+        array = da.copy()
```

Comment:
Just for my understanding--this copies the metadata but not the actual numpy data, correct?

Reply:
Good question, I think actually we might want to pass `deep=False` (see the `.copy` [docstring](https://docs.xarray.dev/en/latest/generated/xarray.DataArray.copy.html))

Reply:
Uh yeah we definitely do not want to load data into memory here! I always assumed the default was deep=False.